internal/controller/apiextensions/composite/composed.go

```diff
@@ -181,15 +181,15 @@ func AssociateByOrder(t []v1.ComposedTemplate, r []corev1.ObjectReference) []Tem
 
 // A CompositionTemplateAssociator returns an array of template associations.
 type CompositionTemplateAssociator interface {
-	AssociateTemplates(context.Context, resource.Composite, *v1.Composition) ([]TemplateAssociation, error)
+	AssociateTemplates(context.Context, resource.Composite, []v1.ComposedTemplate) ([]TemplateAssociation, error)
```

Comment:
This refactoring to operate on `[]v1.ComposedTemplate` is spurious. I had originally thought this would help with this PR because both compositions and revisions have an identical `ComposedTemplate` type. I neglected to realise that the `CompositionRevision` type is different though, due to being in a different package.

I decided to leave the refactoring in because these types don't actually operate on the whole `*v1.Composition` - just the templates. Ideally I'd break it out into its own commit but it got a little bit mixed up with the actual `CompositionRevision` implementation so I'd prefer to just leave it as is if no one minds too much. 😬